File: classwork/18.10.2022/python/main.py
"""
Autors - Kristiāns Francis Cagulis, kc22015
AuPLa0702.
Programma izveidota: 16.10.2022.
"""
import sys
from num_len import get_number_length


def main():
    """Main function"""
    while True:
        user_input = input("Enter numbers: ").lower().split()
        if "q" in user_input:
            sys.exit("User closed the program")
        try:
            numbers = list(map(int, user_input))
        except ValueError:
            print("All entered values were not integers")

        number_length = []
        for number in numbers:
            number_length.append(get_number_length(number))
        print(" ".join(list(map(str, number_length))))


if __name__ == "__main__":
    main()

File: classwork/18.10.2022/cpp/main.cc
/******************************************
Autors - Kristiāns Francis Cagulis, kc22015
AuPla0701.
Programma izveidota: 17.10.2022.
******************************************/

#include "./cipSkaits.hh"
#include "iostream"
using namespace std;

int main() {
  int continueLoop = 1;
  do {
    int arrayLength;
    do {
      cout << "Enter array length: " << endl;
      cin >> arrayLength;
      if (0 < arrayLength && arrayLength <= 100)
        break;
      cout << "Entered number must be smaller or equal than 100" << endl;
    } while (1);
    int numbers[arrayLength];
    int numbersLength[arrayLength];
    cout << "Enter array elements" << endl;
    for (int i = 0; i < arrayLength; i++) {
      cin >> numbers[i];
    }
    for (int i = 0; i < arrayLength; i++) {
      numbersLength[i] = cipSkaits(numbers[i]);
    }
    for (int i = 0; i < arrayLength; i++) {
      cout << numbersLength[i] << " ";
    }

    cout << endl << "Continue? [1/0] ";
    cin >> continueLoop;
  } while (continueLoop);
  return 0;
}
